vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/test/java/com/vaadin/flow/component/sidenav/tests/SideNavItemTest.java

```diff
@@ -437,47 +441,236 @@ public void createFromComponent_setQueryParameters_updateQueryParameters_pathIsU
+    }
+
+    @Test
+    public void setMultiplePathAliases_setEmptyPathAliases_pathAliasesEmpty() {
+        final SideNavItem item = new SideNavItem("Test");
+        item.setPathAliases(Set.of("alias1", "alias2"));
+        item.setPathAliases(Collections.emptySet());
+
+        Assert.assertEquals(0, item.getPathAliases().size());
+        Assert.assertNull(item.getElement().getProperty("pathAliases"));
+    }
+
+    @Test
+    public void setMultiplePathAliases_setPathAliasesNull_pathAliasesEmpty() {
+        final SideNavItem item = new SideNavItem("Test");
+        item.setPathAliases(Set.of("alias1", "alias2"));
+        item.setPathAliases(null);
+
+        Assert.assertEquals(0, item.getPathAliases().size());
+        Assert.assertNull(item.getElement().getProperty("pathAliases"));
+    }
+
+    @Test
+    public void setPathAsComponent_setNullAsComponent_pathAliasesEmpty() {
+        runWithMockRouter(() -> {
+            sideNavItem.setPath(TestRouteWithAliases.class);
+            sideNavItem.setPath((Class<Component>) null);
+
+            Assert.assertEquals(0, sideNavItem.getPathAliases().size());
+            Assert.assertNull(
+                    sideNavItem.getElement().getProperty("pathAliases"));
+        }, TestRouteWithAliases.class);
+    }
+
+    @Test
+    public void setPathAsComponent_setOtherPathAsComponent_pathUpdated() {
+        runWithMockRouter(() -> {
+            sideNavItem.setPath(TestRouteWithAliases.class);
+            sideNavItem.setPath(OtherTestRouteWithAliases.class);
+
+            Assert.assertEquals("bar/foo", sideNavItem.getPath());
+        }, TestRouteWithAliases.class, OtherTestRouteWithAliases.class);
+    }
+
+    @Test
+    public void setPathAsComponent_setOtherPathAsComponent_pathAliasesUpdated() {
+        runWithMockRouter(() -> {
+            sideNavItem.setPath(TestRouteWithAliases.class);
+            sideNavItem.setPath(OtherTestRouteWithAliases.class);
+
+            Assert.assertEquals(1, sideNavItem.getPathAliases().size());
+            Assert.assertNull(sideNavItem.getElement().getProperty("baz/foo"));
+        }, TestRouteWithAliases.class, OtherTestRouteWithAliases.class);
+    }
+
+    @Test
+    public void setPathAsComponent_aliasWithMissingParameterNotAdded() {
+        runWithMockRouter(() -> {
+            sideNavItem.setPath(TestRouteWithAliases.class);
+
+            Assert.assertEquals(2, sideNavItem.getPathAliases().size());
+            Assert.assertEquals(Set.of("foo/baz", "foo/qux"),
+                    sideNavItem.getPathAliases());
+            Assert.assertTrue(Set
+                    .of("[\"foo/baz\",\"foo/qux\"]",
+                            "[\"foo/qux\",\"foo/baz\"]")
+                    .contains(sideNavItem.getElement()
+                            .getProperty("pathAliases")));
+        }, TestRouteWithAliases.class);
+    }
+
+    @Test(expected = NotFoundException.class)
+    public void setPathAsComponent_throws() {
```

Comment:
It's currently not clear from the test name what is expected to throw and why. If `setPath` throws, do we need any following assertions?

Reply:
Good catch. I updated the name and removed the extra assertions.

Reply:
Great, thanks!